lib/commands/connect/sf-auth.js

```diff
@@ -31,24 +38,30 @@ module.exports = {
   needsApp: true,
   needsAuth: true,
   run: cli.command(co.wrap(function* (context, heroku) {
-    let connection = yield api.withConnection(context, heroku);
-    callbackServer();
-
-    let url = '/api/v3/connections/' + connection.id + '/authorize_url';
-    let args = {
-      'environment':'production',
-      // Redirect to the local server created in callbackServer(), so the CLI
-      // can respond immediately after successful authorization
-      'next': `http://localhost:${LOCAL_PORT}`
-    };
-    if (context.flags.login) {
-      args.login_url = context.flags.login;
-    }
-
-    let response = yield api.request(context.auth.password, 'POST', url, args);
-    let redir = response.json.redirect;
-
-    console.log("Launching Salesforce for authorization. If your browser doesn't open, please copy the following URL to proceed:\n\n" + redir + "\n");
-    yield cli.open(redir);
+    let redir;
+
+    yield cli.action('fetching authorizing URL', co(function* () {
+      let connection = yield api.withConnection(context, heroku);
+
+      let url = '/api/v3/connections/' + connection.id + '/authorize_url';
+      let args = {
+        'environment':'production',
```

Comment:
I had wondered about that as well. We can either add all the flags now or push this out and increment later. I had been leaning toward the latter, but I could be persuaded that it's worth the effort now.